pkg/sns/topic_test.go

```diff
@@ -4,36 +4,29 @@ import (
 	"context"
 	"github.com/applike/gosoline/pkg/cfg"
 	gosoAws "github.com/applike/gosoline/pkg/cloud/aws"
-	gosoAwsMocks "github.com/applike/gosoline/pkg/cloud/aws/mocks"
 	monMocks "github.com/applike/gosoline/pkg/mon/mocks"
 	"github.com/applike/gosoline/pkg/sns"
 	snsMocks "github.com/applike/gosoline/pkg/sns/mocks"
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
 	awsSns "github.com/aws/aws-sdk-go/service/sns"
 	"github.com/pkg/errors"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
 	"testing"
 )
 
-func TestTopic_Publish(t *testing.T) {
-	logger := monMocks.NewLoggerMockedAll()
-
-	input := &awsSns.PublishInput{
-		TopicArn: aws.String("arn"),
-		Message:  aws.String("test"),
-	}
-
-	exec := gosoAws.NewTestableExecutor([]gosoAws.TestExecution{{
-		Output: nil,
-		Err:    nil,
-	}})
+type TopicTestSuite struct {
+	suite.Suite
+	client   *snsMocks.Client
+	executor *gosoAws.TestableExecutor
+	topic    *sns.SnsTopic
```

Comment:
Can this be an interface, so SnsTopic does not need to be public?

Reply:
see below